query/node.go

```diff
@@ -101,6 +144,58 @@ type functionExpression struct {
 	groupByCollapses bool
 }
 
+// QueryString does the heavy lifting so implementations don't have to.
+func (f functionExpression) QueryString() string {
+	switch f.functionName {
+	case "+", "-", "*", "/":
+		if len(f.arguments) != 2 {
+			// Then it's not actually an operator.
+			break
+		}
+		return fmt.Sprintf("(%s %s %s)", f.arguments[0].QueryString(), f.functionName, f.arguments[1].QueryString())
+	}
+	argumentQueries := make([]string, len(f.arguments))
+	for i := range argumentQueries {
+		argumentQueries[i] = f.arguments[i].QueryString()
+	}
+	argumentString := strings.Join(argumentQueries, ", ")
+	groupString := ""
+	if len(f.groupBy) != 0 {
+		groupKeyword := "group by"
+		if f.groupByCollapses {
+			groupKeyword = "collapse by"
+		}
+		groupString = fmt.Sprintf(" %s %s", groupKeyword, strings.Join(f.groupBy, ", "))
+	}
+	return fmt.Sprintf("%s(%s%s)", f.functionName, argumentString, groupString)
+}
+func (f functionExpression) Name() string {
+	// TODO: deprecate this behavior before it becomes permanent
+	if f.functionName == "transform.alias" && len(f.arguments) == 2 {
+		log.Printf("I'm an alias")
+		if alias, ok := f.arguments[1].(*stringExpression); ok {
+			return alias.value
+		}
+	}
+	return f.QueryString()
+}
+
+type annotationExpression struct {
+	content    function.Expression
+	annotation string
+}
+
+func (ae annotationExpression) Evaluate(context *function.EvaluationContext) (function.Value, error) {
+	return ae.content.Evaluate(context)
+}
+
+func (ae annotationExpression) QueryString() string {
+	return fmt.Sprintf("%s {%s}", ae.content.QueryString(), ae.annotation)
```

Comment:
The contents of the annotation are always literal, so there's no need to escape.

This is because the annotation provides no way of escaping to begin with. In principle, the only illegal character is `}` and I didn't provide any way of escaping it in the parser. If we wanted to add one then this needs to change- but I don't think there's a significant benefit to allowing escaped characters for something that's supposed to be human-readable.